eng/pipelines/templates/steps/checkout-source-dependencies.yml

```diff
@@ -0,0 +1,32 @@
+parameters:
+  - name: ProjectList
+    type: string
+  - name: CheckoutRecordings
+    type: boolean
+    default: false
+
+steps:
+  - task: PythonScript@0
+    displayName: 'Generate directories variable for sparse checkout'
+    inputs:
+      scriptPath: 'eng/scripts/generate_from_source_pom.py'
+      arguments: '--set-pipeline-variable --project-list ${{ parameters.ProjectList }}'
+      workingDirectory: '$(System.DefaultWorkingDirectory)'
+
+  - pwsh: |
+      $paths = ('$(CheckoutDirectories)' | ConvertFrom-Json) -join ' '
+      $javaPaths = $paths | ForEach-Object { $_ + '/**/*.java' }
+      Invoke-Expression -Command "git sparse-checkout set eng '**/*.xml' '**/*.md' $javaPaths"
```

Comment:
Why have all xml and md files but only some of the java files?

Reply:
We could probably narrow down the xml files to `pom*.xml`, and as well identify the specific markdown files. But they are so cheap to checkout, I figured it was simpler to glob download them, that way we don't have to do too much maintenance should the dependencies on those filetypes change slightly. We also have dependencies in some of the jobs on these filetypes that may exist outside the service directory. The java files in the repo can take up a sizable amount of checkout time (mostly mgmt libraries), so it's definitely something we want to restrict to the service directory only.